.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,74 @@
+name: CI
+
+on:
+  push:
+    branches-ignore:
+      - 'main'
+      - '_bors*'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  selection:
+    runs-on: hetzner-pm
+    container:
+      image: xaynetci/yellow:v1
+    outputs:
+      dart: ${{ steps.filter.outputs.dart }}
+      rust: ${{ steps.filter.outputs.rust }}
+      flutter: ${{ steps.filter.outputs.flutter }}
+    timeout-minutes: 5
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      # For pull requests it's not necessary to checkout the code
```

Comment:
I don't understand this comment because in the previous step we are checking out the code. maybe we can remove this line or check if this action works without checking out the repo first

Reply:
Fair, I don't understand it either tbh 🤔 I'll take a look.